src/NeinLinq.EntityFrameworkCore/EntityQueryCompilerAdapter.cs

```diff
@@ -29,11 +29,18 @@ public Func<QueryContext, TResult> CreateCompiledQuery<TResult>(Expression query
     public Func<QueryContext, TResult> CreateCompiledAsyncQuery<TResult>(Expression query)
         => innerCompiler.CreateCompiledQuery<TResult>(RewriteQuery(query));
 
+#if NET9_0_OR_GREATER
```

Comment:
It's possible to have .NET 9 DI/EFCore nuget packages on .NET 8. (minimal version). Perhaps make it NET8_0_OR_GREATER?

Reply:
But that would *require* to use EF 9 on .NET 8...

Reply:
But this can do it - I don't think pre-compiled queries will work w/ lambda injection anyway...